### PR TITLE
Backup code improvements in login portal

### PR DESCRIPTION
**Pull request by bhagyasakalanka**
### Related PRs
- https://github.com/wso2/identity-apps/pull/2963
- https://github.com/wso2/identity-apps/pull/2964

Need to merge following PRs before merge this PR.
- https://github.com/wso2/carbon-identity-framework/pull/4009
- https://github.com/wso2/carbon-identity-framework/pull/4010
- https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/416
- https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/1
- https://github.com/wso2/identity-api-user/pull/146

### CODE REVIEW

File: apps/authentication-portal/src/main/webapp/backupCode.jsp [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+<%@ page import="org.owasp.encoder.Encode" %>
+<%@ page import="org.wso2.carbon.identity.application.authentication.endpoint.util.Constants" %>
+<%@ page import="java.io.File" %>
+<%@ page import="java.util.Map" %>
+<%@ page language="java" contentType="text/html; charset=UTF-8" pageEncoding="UTF-8" %>
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+<%@ include file="includes/localize.jsp" %>
+<%@ include file="includes/init-url.jsp" %>
+
+    <% request.getSession().invalidate(); String queryString=request.getQueryString();
```
Comment (brionmario) [inline apps/authentication-portal/src/main/webapp/backupCode.jsp:17]: Indent line 17:153. Tab space is not needed.

File: apps/authentication-portal/src/main/webapp/backupCode.jsp [hunk 2]
```diff
@@ -0,0 +1,153 @@
+<%-- ~ Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved. ~ ~ WSO2 Inc. licenses this file to you
```
Comment (brionmario) [inline apps/authentication-portal/src/main/webapp/backupCode.jsp:1]: Add the proper license.

File: apps/authentication-portal/src/main/webapp/WEB-INF/web.xml [hunk 3]
```diff
@@ -312,6 +312,16 @@
...
     </servlet>
 
+    <servlet>
+        <servlet-name>backup_code.do</servlet-name>
+        <jsp-file>/backupCode.jsp</jsp-file>
+    </servlet>
+
+    <servlet>
+        <servlet-name>backup_code_error.do</servlet-name>
+        <jsp-file>/backupCodeError.jsp</jsp-file>
```
Comment (brionmario) [inline apps/authentication-portal/src/main/webapp/WEB-INF/web.xml:322]: Verify the file name convention. AFAIR, we have `-` separated file names.

File: apps/authentication-portal/src/main/webapp/backupCode.jsp (hunk 2 above)
Comment (brionmario) [reply apps/authentication-portal/src/main/webapp/backupCode.jsp:1]: https://github.com/wso2/identity-apps/blob/master/apps/authentication-portal/src/main/webapp/domain.jsp#L1

**Review by brionmario — APPROVED** [review]

File: apps/authentication-portal/src/main/webapp/WEB-INF/web.xml (hunk 3 above)
Comment (bhagyasakalanka) [reply apps/authentication-portal/src/main/webapp/WEB-INF/web.xml:322]: In some places we used `-` but in some places we used camel case. But most of the cases, `-` is used. So I'll update the file names with `-`

**Pull request merged**